### PR TITLE
fix(docs): docs hyperlink behaviour issues for unlink button and popover

### PR DESCRIPTION
**Pull request by Gopi-bruno**
BRU-4233 | BRU-4255

### Description

Fixes the hyperlink functionality issues. 
- Remove icon functions correctly now.
- Redirect icon is removed and the link in the hover popover is clickable now.
- Removed `Remove` button is the Edit Popover.
- Added e2e tests to ensure these functionalities are tested with playwright.

#### Before
Screenshot 2026-08-10 at 4 23 54 PM Screenshot 2026-08-10 at 4 23 44 PM
#### After
Screenshot 2026-08-10 at 4 22 48 PM
Screenshot 2026-08-10 at 4 22 57 PM

#### Contribution Checklist:

- [x] **I've used AI significantly to create this pull request**
- [x] **The pull request only addresses one issue or adds one feature.**
- [x] **The pull request does not introduce any breaking changes**
- [x] **I have added screenshots or gifs to help explain the change if applicable.**
- [x] **I have read the [contribution guidelines](https://github.com/usebruno/bruno/blob/main/contributing.md).**
- [x] **Create an issue and link to the pull request.**
- [x] **I've run the claude code review skill locally.**

Note: Keeping the PR small and focused helps make it easier to review and merge. If you have multiple changes you want to make, please consider submitting them as separate pull requests.

#### Publishing to New Package Managers

Please see [here](../publishing.md) for more information.

## Summary by CodeRabbit

* **New Features**
  * Improved rich-text link editing, hovering, and unlinking while preserving surrounding text.
  * Link popovers now reposition automatically during scrolling, resizing, and layout changes.
  * Popovers adapt their placement to remain visible within the available viewport.
  * Added smoother hover interactions between links and their popovers.
  * Double-click a document area to enter editing mode when applicable.

* **Bug Fixes**
  * Improved popover visibility and hover navigation across editing modes.
  * Unlinking one link no longer affects other links or surrounding text.
  * Removed the separate “open in new tab” control.

### CODE REVIEW

File: packages/bruno-app/src/ui/RichTextEditor/components/EditorLinkPopover/index.js [hunk 1]
```diff
@@ -31,30 +34,44 @@ function resolveLinkText(editor, anchorEl) {
...
-  return { top, left };
+  // Try bottom first
+  let top = anchorRect.bottom + 4;
+
+  // If it overflows the bottom of the viewport and there is space above, place it above
+  if (top + popoverHeight > window.innerHeight && anchorRect.top - popoverHeight - 4 > 0) {
+    top = anchorRect.top - popoverHeight - 4;
+  }
+
+  return { top, left, visible: isVisible };
```
[the remark this answers is not in the document]
Comment (sachin-thakur-bruno) [reply packages/bruno-app/src/ui/RichTextEditor/components/EditorLinkPopover/index.js]: @Gopi-bruno worth checking this.

File: packages/bruno-app/src/ui/RichTextEditor/components/EditorLinkPopover/index.js [hunk 2]
```diff
@@ -71,7 +88,28 @@ const EditorLinkPopover = ({ editor, onSubmit, onUnlink, containerEl }) => {
...
+
+  const measureHeight = (elRef, lastKnownRef) => {
+    const height = elRef.current?.getBoundingClientRect().height;
+    if (height) lastKnownRef.current = height;
+    return lastKnownRef.current;
+  };
+
+  const [isEditable, setIsEditable] = useState(editor?.isEditable ?? false);
+
+  useEffect(() => {
```
Comment (sachin-thakur-bruno) [inline packages/bruno-app/src/ui/RichTextEditor/components/EditorLinkPopover/index.js]: can we use `const isEditable = editor?.isEditable ?? false;`. which will remove the whole `useEffect` and the `isEditable` state.

File: tests/utils/page/docs/index.ts [hunk 3]
```diff
@@ -27,5 +27,14 @@ export const buildDocsLocators = (page: Page) => ({
   codeBlockLangOption: (language: string) => page.locator(`[role="menuitem"][data-item-id="${language}"]`),
   codeBlockCopyBtn: () => page.getByTestId('code-block-copy-btn'),
   codeBlockContent: () => page.locator('[data-testid="code-block-pre"] code'),
-  codeBlockSyntaxHighlight: (className: string) => page.locator(`[data-testid="code-block-pre"] code .${className}`)
+  codeBlockSyntaxHighlight: (className: string) => page.locator(`[data-testid="code-block-pre"] code .${className}`),
+  // Link popover locators
+  linkEditPopover: () => page.getByTestId('editor-link-popover'),
+  linkEditUrlInput: () => page.locator('#linkUrl'),
+  linkEditInsertBtn: () => page.getByRole('button', { name: 'Insert' }),
+  linkEditSaveBtn: () => page.getByRole('button', { name: 'Save' }),
```
Comment (sachin-thakur-bruno) [inline tests/utils/page/docs/index.ts]: can we Scope the buttons to [data-editor-link-popover] ?

File: tests/utils/page/docs/index.ts [hunk 4]
```diff
@@ -27,5 +27,14 @@ export const buildDocsLocators = (page: Page) => ({
   codeBlockLangOption: (language: string) => page.locator(`[role="menuitem"][data-item-id="${language}"]`),
   codeBlockCopyBtn: () => page.getByTestId('code-block-copy-btn'),
   codeBlockContent: () => page.locator('[data-testid="code-block-pre"] code'),
-  codeBlockSyntaxHighlight: (className: string) => page.locator(`[data-testid="code-block-pre"] code .${className}`)
+  codeBlockSyntaxHighlight: (className: string) => page.locator(`[data-testid="code-block-pre"] code .${className}`),
+  // Link popover locators
+  linkEditPopover: () => page.getByTestId('editor-link-popover'),
+  linkEditUrlInput: () => page.locator('#linkUrl'),
```
Comment (sachin-thakur-bruno) [inline tests/utils/page/docs/index.ts]: can we use data-testid/role.

File: tests/richtext-editor/links.spec.ts [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+test.describe('Rich Text Editor Edge Cases - Links', () => {
+  test.beforeEach(async ({ page }) => {
+    await page.setViewportSize({ width: 1920, height: 1080 });
+  });
+
+  test.afterEach(async ({ page }) => {
+    await closeAllCollections(page);
+  });
+
+  test('Creating, editing, and removing links', async ({ page, createTmpDir }) => {
```
Comment (sachin-thakur-bruno) [inline tests/richtext-editor/links.spec.ts:11]: can we please use `test.step` wrappers ?

File: packages/bruno-app/src/ui/RichTextEditor/components/EditorLinkPopover/StyledWrapper.js [hunk 6]
```diff
@@ -10,6 +10,15 @@ const StyledWrapper = styled.div`
   border-radius: ${(props) => props.theme.border.radius.base};
   box-shadow: ${(props) => props.theme.dropdown.shadow || 'none'};
 
+  &::before {
```
Comment (sachin-thakur-bruno) [inline packages/bruno-app/src/ui/RichTextEditor/components/EditorLinkPopover/StyledWrapper.js:13]: do we need this?

File: packages/bruno-app/src/ui/RichTextEditor/components/EditorLinkEditPopover/index.js [hunk 7]
```diff
@@ -2,12 +2,17 @@ import React, { useState, useEffect, useRef } from 'react';
...
 import StyledWrapper from './StyledWrapper';
 
-const EditorLinkEditPopover = ({ isOpen, onClose, onSubmit, onUnlink, initialText, initialUrl, externalCoords }) => {
+const EditorLinkEditPopover = ({ isOpen, onClose, onSubmit, initialText, initialUrl, externalCoords, wrapperRef }) => {
   const [text, setText] = useState(initialText || '');
   const [url, setUrl] = useState(initialUrl || '');
   const popoverRef = useRef(null);
   const urlInputRef = useRef(null);
 
+  const setPopoverRefs = (node) => {
```
Comment (sachin-thakur-bruno) [inline packages/bruno-app/src/ui/RichTextEditor/components/EditorLinkEditPopover/index.js]: nit: can we use ` useCallback` for this? else it will re-attaches on every render.

File: packages/bruno-app/src/ui/RichTextEditor/components/EditorLinkPopover/index.js [hunk 8]
```diff
@@ -146,6 +189,61 @@ const EditorLinkPopover = ({ editor, onSubmit, onUnlink, containerEl }) => {
     };
   }, [editor, getContainer]);
 
+  useEffect(() => {
+    if (!editor) return;
```
Comment (sachin-thakur-bruno) [inline packages/bruno-app/src/ui/RichTextEditor/components/EditorLinkPopover/index.js]: can we also add a gate for (!hoverOpen && !editOpen) ?

File: packages/bruno-app/src/ui/RichTextEditor/components/EditorLinkPopover/StyledWrapper.js (hunk 6 above)
Comment (Gopi-bruno) [reply packages/bruno-app/src/ui/RichTextEditor/components/EditorLinkPopover/StyledWrapper.js:13]: Yes, we will need this to maintain the hover popover smooth. Without this, it would break the free movement between the link text and the hover popover.

File: packages/bruno-app/src/ui/Portal/index.js [hunk 9]
```diff
@@ -0,0 +1,7 @@
+import { createPortal } from 'react-dom';
+
+function Portal({ children }) {
```
Comment (sid-bruno) [inline packages/bruno-app/src/ui/Portal/index.js]:
```suggestion
function Portal({ children, root = document.body }) {
```

File: packages/bruno-app/src/ui/Portal/index.js [hunk 10]
```diff
@@ -0,0 +1,7 @@
...
+  return createPortal(children, document.body);
```
Comment (sid-bruno) [inline packages/bruno-app/src/ui/Portal/index.js]:
```suggestion
  return createPortal(children, root)
```

File: packages/bruno-app/src/ui/RichTextEditor/components/EditorLinkEditPopover/index.js [hunk 11]
```diff
@@ -1,13 +1,18 @@
-import React, { useState, useEffect, useRef } from 'react';
+import React, { useState, useEffect, useRef, useCallback } from 'react';
 import Button from 'ui/Button';
 import StyledWrapper from './StyledWrapper';
 
-const EditorLinkEditPopover = ({ isOpen, onClose, onSubmit, onUnlink, initialText, initialUrl, externalCoords }) => {
+const EditorLinkEditPopover = ({ isOpen, onClose, onSubmit, initialText, initialUrl, externalCoords, wrapperRef }) => {
```
Comment (sid-bruno) [inline packages/bruno-app/src/ui/RichTextEditor/components/EditorLinkEditPopover/index.js]: any specific reason to not use `forwardRef`?

File: packages/bruno-app/src/ui/RichTextEditor/components/EditorLinkPopover/index.js
Comment (sid-bruno) [inline packages/bruno-app/src/ui/RichTextEditor/components/EditorLinkPopover/index.js:1]: we already have `@popperjs/core@2.11.8` (`@tippyjs/react` → `tippy.js` → `@popperjs/core`)  as a transitive dep, we can just use that for the positioning calculation instead of hand coding the logic

**Review by sid-bruno — DISMISSED** [review]

**Comment by sid-bruno** [comment]
CRUD e2e in `tests/richtext-editor/links.spec.ts` covers create / edit / unlink, but not the portal + fixed-position

Please add e2e coverage for:

1. **Scroll hide** - open the hover (or edit) popover on a link, scroll the docs pane until the anchor leaves the editor container, assert the popover is not visible (`display: none` / not shown).
2. **Scroll track** - with the popover open and the anchor still in view, scroll a bit and assert the popover stays anchored near the link (fixed coords update).
3. **Flip (optional but useful)** - place a link near the bottom of the viewport so the popover would overflow, assert it renders above the anchor (`$placement === 'top'` / top < anchor top).

Without these, the hand-rolled `getPortalCoords` + scroll/resize/`ResizeObserver` path can regress with a green CRUD suite.

File: packages/bruno-app/src/ui/RichTextEditor/components/EditorLinkEditPopover/index.js (hunk 11 above)
Comment (Gopi-bruno) [reply packages/bruno-app/src/ui/RichTextEditor/components/EditorLinkEditPopover/index.js]: using forwardRef now

File: packages/bruno-app/src/ui/RichTextEditor/components/EditorLinkPopover/index.js
Comment (Gopi-bruno) [reply packages/bruno-app/src/ui/RichTextEditor/components/EditorLinkPopover/index.js:1]: using `@popperjs/core` now for the positioning calculation, and it is added as a direct dependency in the `package.json`

**Comment by Gopi-bruno** [comment]
> CRUD e2e in `tests/richtext-editor/links.spec.ts` covers create / edit / unlink, but not the portal + fixed-position
> 
> Please add e2e coverage for:
> 
> 1. **Scroll hide** - open the hover (or edit) popover on a link, scroll the docs pane until the anchor leaves the editor container, assert the popover is not visible (`display: none` / not shown).
> 2. **Scroll track** - with the popover open and the anchor still in view, scroll a bit and assert the popover stays anchored near the link (fixed coords update).
> 3. **Flip (optional but useful)** - place a link near the bottom of the viewport so the popover would overflow, assert it renders above the anchor (`$placement === 'top'` / top < anchor top).
> 
> Without these, the hand-rolled `getPortalCoords` + scroll/resize/`ResizeObserver` path can regress with a green CRUD suite.

Added tests for these cases

**Review by sid-bruno — DISMISSED** [review]

**Review by sachin-thakur-bruno — APPROVED** [review]

**Review by sid-bruno — APPROVED** [review]
